apis/datadoghq/v1alpha1/datadogautoscaling_types.go

```diff
@@ -0,0 +1,282 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package v1alpha1
+
+import (
+	autoscalingv2 "k8s.io/api/autoscaling/v2"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// spec:
+//   targetRef:
+//     apiVersion: apps/v1
+//     kind: Deployment
+//     name: test
+//   owner: local
+//   policy:
+//     update:
+//       mode: Auto|DryRun
+//   recommender:
+//     name: containerMetrics
+//     containerMetrics:
+//       cpuUtilizationTarget: 60
+//   constraints:
+//     minReplicas: 1
+//     maxReplicas: 10
+//     containers:
+//       - name: "*"
+//         enabled: true
+//         requests:
```

Comment:
Should we have a parent key to indicate to what resource type this applies to? I guess it's memory here.